pkg/morph/client/multi.go

```diff
@@ -15,21 +15,28 @@ type Endpoint struct {
 // SwitchRPC performs reconnection and returns true if it was successful.
 func (c *Client) SwitchRPC() bool {
 	c.switchLock.Lock()
-	defer c.switchLock.Unlock()
 
 	for attempt := 0; attempt < c.cfg.reconnectionRetries; attempt++ {
 		if c.switchPRC() {
+			c.switchLock.Unlock()
+
+			if c.cfg.rpcSwitchCb != nil {
+				c.cfg.rpcSwitchCb()
+			}
+
 			return true
 		}
 
 		select {
 		case <-time.After(c.cfg.reconnectionDelay):
 		case <-c.closeChan:
+			c.switchLock.Unlock()
 			return false
 		}
 	}
 
 	c.inactive = true
+	c.switchLock.Unlock()
```

Comment:
placing this before statemenet below changes method behavior, i'd do it in a separate commit with explanation (i realize why u've done it). And then autonomous commit regarding the new callback